### DISCUSSION TITLE
How do I get a template rendering value or string？

### DISCUSSION THREAD

**Discussion by zgangguoguo** [rwf2/Rocket#2226] [Questions] [answered]
I want to cache the template rendering results with Redis, is there any way to get the HTLM string from the template rendering?

**Comment by wezm** [comment, marked as answer]
I think [Template::show](https://api.rocket.rs/v0.5-rc/rocket_dyn_templates/struct.Template.html#method.show) is what you're after.

**Reply by zgangguoguo** [reply]
> I think [Template::show](https://api.rocket.rs/v0.5-rc/rocket_dyn_templates/struct.Template.html#method.show) is what you're after.

In rocket API:   **Render the template named name with the context context into a String. This method should not be used in any running Rocket application. This method should only be used during testing to validate Template responses. For other uses, use [render()](https://api.rocket.rs/v0.5-rc/rocket_dyn_templates/struct.Template.html#method.render) instead.**

It does not appear to be recommended for use in production environments

**Reply by wezm** [reply]
Oh whoops, sorry I didn't see that bit. I'm not sure how else to do what you're after.

**Reply by zgangguoguo** [reply]
> Oh whoops, sorry I didn't see that bit. I'm not sure how else to do what you're after.

Yesh, it's a bit difficult. But I still want to thank you for your help.

**Reply by SergioBenitez** [reply]
I think you're after https://api.rocket.rs/master/rocket_dyn_templates/struct.Metadata.html#method.render, which will be in the 0.5.0 release shortly.

**Reply by zgangguoguo** [reply]
> I think you're after https://api.rocket.rs/master/rocket_dyn_templates/struct.Metadata.html#method.render, which will be in the 0.5.0 release shortly.

Yes, it's probably just what I need. I'm really looking forward to the release of the new version。
